lib/mongoid/extensions/binary.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Mongoid
+  module Extensions
+    module Binary
+
+      # Turn the object from the ruby type we deal with to a Mongo friendly
+      # type.
+      #
+      # @example Mongoize the object.
+      #   object.mongoize
+      #
+      # @return [ Object ] The object.
+      def mongoize
+        BSON::Binary.mongoize(self)
+      end
+
+      module ClassMethods
+
+        # Mongoize an object of any type to how it's stored in the db.
+        #
+        # @example Mongoize the object.
+        #   BigDecimal.mongoize(123)
+        #
+        # @param [ Object ] object The object to Mongoize
+        #
+        # @return [ String | BSON::Binary | nil ] A String or Binary
+        #   representing the object or nil.
+        def mongoize(object)
+          return if object.nil?
+
+          case object
+          when BSON::Binary
+            object
+          when String
+            BSON::Binary.new(object)
+          else
+            # TODO: MONGOID-5222 raise on the setting of feature flag.
```

Comment:
What are your thoughts on treating Symbol as String here?

Reply:
I would be okay with that